presto-main/src/main/java/com/facebook/presto/server/protocol/PrefixUrlUtil.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.server.protocol;
+
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.Response;
+
+import java.net.URL;
+
+import static com.facebook.presto.client.PrestoHeaders.PRESTO_PREFIX_URL;
+import static javax.ws.rs.core.MediaType.TEXT_PLAIN_TYPE;
+
+public class PrefixUrlUtil
+{
+    private PrefixUrlUtil() {}
+
+    public static void abortIfPrefixUrlInvalid(String xPrestoPrefixUrl)
```

Comment:
Can we move this to `QueryResourceUtil`?

Reply:
Sure. I've done that now and got rid of the utility class. `abortIfPrefixUrlInvalid` is now a static method of `QueryResourceUtil`